apps/wallet-dashboard/components/AssetsList.tsx

```diff
@@ -4,10 +4,17 @@
 import { AssetCategory } from '@/lib/enums';
 import { IotaObjectData } from '@iota/iota-sdk/client';
 import { AssetTileLink } from '@/components';
+import { LoadingIndicator } from '@iota/apps-ui-kit';
+import { useEffect, useRef } from 'react';
+import { useOnScreen } from '@iota/core';
+import cl from 'clsx';
 
 interface AssetListProps {
     assets: IotaObjectData[];
     selectedCategory: AssetCategory;
+    hasNextPage?: boolean;
+    isFetchingNextPage?: boolean;
+    fetchNextPage?: () => void;
```

Comment:
These are never optional
```suggestion
    hasNextPage: boolean;
    isFetchingNextPage: boolean;
    fetchNextPage: () => void;
```